### DISCUSSION TITLE
Winlogbeat Encryption to Security Onion

### DISCUSSION THREAD

**Discussion by ColeVan** [Security-Onion-Solutions/securityonion#2503] [Unsupported Versions] [answered]
Hopefully someone can give me clear instruction on how to accomplish this. 
https://docs.securityonion.net/en/2.3/beats.html

After following the steps in the security onion documentation above I am unable to establish an encrypted connection between winlogbeats on the endpoint and our sec onion instance. Below are snapshots of what my 009 beats.conf looks like, endpoint winlogbeat.yml file, pcap file after initiating winlogbeats , and what certs I am using. The beats.conf file is different from what documentation illustrates but I tried following it both ways and still no luck.

Am I missing something? I am sure that others have solved this problem before. Any help would much appreciated at this point. FYSA: I used the filebeat cert because I have no idea how to make a beat cert. If I am required to make a new beat cert how is that accomplished?  

![image](https://user-images.githubusercontent.com/70167373/103384710-5e894800-4ac5-11eb-9b0d-e8e7a388d89f.png)
![image](https://user-images.githubusercontent.com/70167373/103384698-54ffe000-4ac5-11eb-907e-6c0b08a4efd2.png)
![image](https://user-images.githubusercontent.com/70167373/103384674-4285a680-4ac5-11eb-875f-a20205c2959f.png)
![image](https://user-images.githubusercontent.com/70167373/103384632-14a06200-4ac5-11eb-9169-d3c810f4cc60.png)
![image](https://user-images.githubusercontent.com/70167373/103384557-d145f380-4ac4-11eb-914e-542236d5693e.png)

**Comment by weslambert** [comment, marked as answer]
You'll want to reference the paths as described in the docs (`/usr/share/...`), as the container knows not of the local filesystem path, but the path mapped to the container (see [here](https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/logstash/init.sls#L176)).

Ex.

```
ssl_certificate_authorities => ["/usr/share/logstash/myca.crt"]
ssl_certificate => "/usr/share/logstash/certs/mybeats.crt"
ssl_key => "/usr/share/logstash/certs/mybeats.key"
```

Also, I just updated the path to which you should copy the certs on the manager -- should be `/opt/so/conf/logstash/etc/certs`, as opposed to `/opt/so/conf/logstash/certs`.

**Reply by ColeVan** [reply]
Thank you. I copied the certs to the new directory and restarted logstash. Getting a new error now. Any thoughts?

![image](https://user-images.githubusercontent.com/70167373/103552192-838d0a80-4e79-11eb-853f-624f27c17628.png)

Update:
Once you copy the certs over to their respective directory and restart logstash they disappear. Also, shouldn't the 0009beat.conf reflect in the logstash docker?  

Update: Here are the screenshots of what it looks like when you cp the certs to the correct directory and restart logstash. The certs disappear.
![image](https://user-images.githubusercontent.com/70167373/103651558-cdd0c300-4f2f-11eb-937a-02af19cec7a4.png)
![image](https://user-images.githubusercontent.com/70167373/103651632-e80aa100-4f2f-11eb-96ab-7ad8b8d15d2d.png)

Update:

Finally figured it out. Some of your documentation is wrong on where to copy the beats.conf to. I will document and post to this chat today.

**Reply by pigeon-tiger** [reply]
@ColeVan I'm having the same problem - disappearing certs. What was the fix/change you mention you found in your last update?

**Reply by ColeVan** [reply]
1.)	Copy the 0009_inputs_beats.conf using the command below
cp /opt/so/saltstack/default/salt/logstash/pipeline/config/so/0009_input_beats.conf /opt/so/saltstack/local/salt/logstash/pipelines/config/so/

2.)	Change directory where the 0009_ input_beats.conf is now located and configure it with the test below.
cd /opt/so/saltstack/local/salt/logstash/pipelines/config/so/
vi 0009_ input_beats.conf
input {
  beats {
    port => "5044"
    ssl => true
    ssl_certificate_authorities => ["/usr/share/filebeat/ca.crt"]
    ssl_certificate => "/usr/share/logstash/filebeat.crt"
    ssl_key => "/usr/share/logstash/filebeat.key"
    tags => [ "beat-ext" ]
  }
}

3.)	Once the 0009_ input_beats.conf is configured make sure to copy the proper certs from /etc/pki/ to  /opt/so/conf/logstash/etc/certs/ 

4.)	The init.sls file will show the paths in which sec onion maps the certs to docker
Note: There are issues when creating new certs and conducting the steps 1-4. After logstash restarts errors in the logs outline that new certs have no valid paths. Unsure why this occurs but the init.sls file successfully copies over new certs when they are copied to /opt/so/conf/logstash/etc/certs/. 



These are just a few steps I took in the right direction when using the filebeat cert and key. I can send you the word doc I used to document how to setup winlogbeat encryption. What's your email?

**Reply by pigeon-tiger** [reply]
Thanks for the response, that's an approach I didn't think of at all. Later yesterday I got another method to work based on combining the docs and another discussion. Simply placed the certs and key in 

/opt/so/saltstack/local/salt/logstash/etc/certs

which then placed them into /opt/so/conf/logstash/etc/certs/ once I restarted logstash, and consequentially binded them to the container paths from the beats.conf

**Reply by jaredatobe** [reply]
Documentation has now been updated with the [correct path](https://github.com/Security-Onion-Solutions/securityonion/discussions/4964).
